### DISCUSSION TITLE
Programación y videojuegos

### DISCUSSION THREAD

**Discussion by A-Man6** [hackatomik/SISCOx256#50] [Chill | Carrera]
En lo personal he tenido curiosidad en saber de qué forma es que se logran programar los videojuegos que conocemos. ¿Cuáles son las ventajas que tienen ciertos motores sobre otros como unreal y unity? y ¿Cómo es que se logra el crossplay en línea al tener diferentes dispositivos?

**Comment by quantumquirkxyz** [comment]
Para programar los videojuegos que conocemos, se utilizan motores de juego como Unreal Engine y Unity, cada uno con sus propias ventajas. Unreal Engine es conocido por sus gráficos de alta calidad y su potente sistema de scripting visual, lo que es ideal para crear juegos con gráficos detallados y complejos. Unity, por otro lado, es muy versátil y fácil de usar, lo que lo hace popular para una variedad de juegos, desde 2D hasta 3D, y también es muy accesible para principiantes.

El crossplay en línea, que permite que jugadores en diferentes dispositivos jueguen juntos, se logra a través de la implementación de servidores y sistemas de comunicación que permiten la conexión entre plataformas. Los motores de juego suelen tener herramientas y APIs específicas para manejar el crossplay, asegurando que los datos y la sincronización entre los diferentes dispositivos sean coherentes. Esto permite que los jugadores, sin importar el dispositivo que usen, puedan disfrutar del juego juntos sin problemas.

**Reply by quantumquirkxyz** [reply]
Yo recomiendo el uso de motores de videojuegos opensource tipo: 
1. **Godot Engine**: Es un motor de juego muy popular y completamente open source. Ofrece un entorno intuitivo para el desarrollo de juegos en 2D y 3D. Su lenguaje de scripting, GDScript, es fácil de aprender y tiene una comunidad activa.

2. **O3DE (Open 3D Engine)**: Anteriormente conocido como Amazon Lumberyard, O3DE es un motor de juego open source de alto rendimiento enfocado en gráficos de calidad y capacidades avanzadas para desarrollos en 3D.

3. **Torque3D**: Este motor es conocido por su robustez en el desarrollo de juegos en 3D. Es flexible y se puede adaptar a una variedad de tipos de juegos. Aunque su desarrollo se ha desacelerado, sigue siendo una opción válida para proyectos open source.

4. **Panda3D**: Este motor es utilizado principalmente para desarrollar juegos en 3D y simulaciones. Es conocido por su capacidad de scripting en Python, lo que facilita el desarrollo para aquellos que ya están familiarizados con el lenguaje.

5. **Battle for Wesnoth**: Aunque más conocido como un juego, el motor de Wesnoth es open source y puede ser útil para quienes están interesados en el desarrollo de juegos de estrategia por turnos.

n.
